pkg/kncloudevents/message_sender.go

```diff
@@ -72,3 +94,82 @@ func (s *HttpMessageSender) NewCloudEventRequestWithTarget(ctx context.Context,
 func (s *HttpMessageSender) Send(req *nethttp.Request) (*nethttp.Response, error) {
 	return s.Client.Do(req)
 }
+
+// CheckRetry specifies a policy for handling retries. It is called
+// following each request with the response and error values returned by
+// the http.Client. If CheckRetry returns false, the Client stops retrying
+// and returns the response to the caller. If CheckRetry returns an error,
+// that error value is returned in lieu of the error from the request. The
+// Client will close any response body when retrying, but if the retry is
+// aborted it is up to the CheckRetry callback to properly close any
+// response body before returning.
+type CheckRetry func(ctx context.Context, resp *nethttp.Response, err error) (bool, error)
+
+// Backoff specifies a policy for how long to wait between retries.
+// It is called after a failing request to determine the amount of time
+// that should pass before trying again.
+type Backoff func(attemptNum int, resp *nethttp.Response) time.Duration
+
+type RetriesConfig struct {
```

Comment:
This is bit of a nit but still wanted to jot it down as I'm reading through this, as it might be for others that read the code as well. I had a hard time understanding the plurals here, and in signatures using plurals. I think it would be clearer to have this as RetryConfig. There's at least one place where we then take a vararg that itself is a plural, which made things even more confusing (to me at least).

Reply:
Same thought while reading some literature and other implementation about retry, they use `RetryConfig`.

Reply:
+1 for `RetryConfig`

Reply:
Done: https://github.com/knative/eventing/pull/2932/commits/3a435783843a4e79ed0df89622d15ecad376ef1c